api/src/main/java/com/netflix/iceberg/Tables.java

```diff
@@ -43,4 +43,13 @@ Table create(Schema schema,
                String tableIdentifier);
 
   Table load(String tableIdentifier);
+
+  /**
+   * Drop a table from iceberg catalog, the underlying implementation
+   * may or may not drop the corresponding metadata and data.
+   * @param tableIdentifier
+   */
+  void drop(String tableIdentifier);
+
+  void rename(String oldTableIdentifier, String newTableIdentifier);
```

Comment:
How about we provide a flag to create parent directories if necessary? A properties map will be a cleaner future proof contract.